core/fragment.c

```diff
@@ -2369,7 +2369,7 @@ fragment_create(dcontext_t *dcontext, app_pc tag, int body_size, int direct_exit
         if (!fragment_lookup_deleted(dcontext, tag) && !TEST(FRAG_COARSE_GRAIN, flags))
             STATS_INC(num_unique_fragments);
     });
-    if (d_r_stats != NULL &&
+    if (GLOBAL_STATS_ON() &&
```

Comment:
Maybe this PR should add a long-suite test that disables global stats.

Reply:
Done.